### PR TITLE
update for go module compatibility

### PR DESCRIPTION
**Pull request by Xe**
Update internal imports for [Semantic Import Versioning](https://research.swtch.com/vgo-import) compatibility. I realize this is dumb and means existing code needs to be updated or you could import two versions of storm. This is the state of the world with Go modules.

Please make a semver patch bump.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,20 @@
-module github.com/asdine/storm
+module github.com/asdine/storm/v2
 
 require (
+	github.com/DataDog/zstd v1.3.5 // indirect
 	github.com/Sereal/Sereal v0.0.0-20181211220259-509a78ddbda3
+	github.com/asdine/storm v2.1.2+incompatible
```
Comment (jrick) [inline go.mod]: I don't think storm should be depending on an older version of itself after this change.  Did you miss an import change perhaps?

Comment (chappjc) [reply go.mod]: Please merge or reference https://github.com/Xe/storm/pull/1 for the complete fix.

Comment (Xe) [reply go.mod]: thank you @chappjc

**Comment by anjmao** [comment]
I think it's better to close this and merge https://github.com/asdine/storm/pull/256  as v2 is already released so it can break consumers who already uses v2.

**Comment by chappjc** [comment]
I had completely forgotten about this PR, but I agree with @anjmao regarding /v3.  The reasoning is outlined in https://github.com/asdine/storm/issues/255 and the linked docs from there.

**Pull request closed without merging**